src/test/java/ru/hogwarts/school/services/FacultyServiceTest.java

```diff
@@ -0,0 +1,196 @@
+
+    @Test
+    void get() throws Exception {
+        JSONObject jsonObject = new JSONObject();
+        jsonObject.put("name",name);
+        jsonObject.put("color",color);
+        Faculty faculty = new Faculty();
+        faculty.setId(id);
+        faculty.setName(name);
+        faculty.setColor(color);
+        when(facultyRepository.save(any(Faculty.class))).thenReturn(faculty);
+        when(facultyRepository.findById(any(Long.class))).thenReturn(Optional.of(faculty));
+        mockMvc.perform(MockMvcRequestBuilders
+                        .post("/faculty")
+                        .content(jsonObject.toString())
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(id))
+                .andExpect(jsonPath("$.name").value(name))
+                .andExpect(jsonPath("$.color").value(color));
+        when(facultyService.get(id)).thenReturn(faculty);
+        mockMvc.perform(MockMvcRequestBuilders
+                        .get("/faculty/" + id)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(id))
+                .andExpect(jsonPath("$.name").value(name))
+                .andExpect(jsonPath("$.color").value(color));
+    }
+
+    @Test
+    void findFacultyById() throws Exception {
+        create();
+        get();
+        Long studentId = 1L;
+        String studentName = "Garry";
+        int studentAge = 11;
+        JSONObject jsonObject = new JSONObject();
+        jsonObject.put("name",studentName);
+        jsonObject.put("age",studentAge);
+        Student student = new Student();
+        student.setId(studentId);
+        student.setName(studentName);
+        student.setAge(studentAge);
+        when(studentRepository.save(any(Student.class))).thenReturn(student);
+        when(studentRepository.findById(any(Long.class))).thenReturn(Optional.of(student));
+        mockMvc.perform(MockMvcRequestBuilders
+                        .post("/student")
+                        .content(jsonObject.toString())
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(studentId))
+                .andExpect(jsonPath("$.name").value(studentName))
+                .andExpect(jsonPath("$.age").value(studentAge));
+        when(facultyService.findFacultyById(id)).thenReturn(List.of(student));
+        mockMvc.perform(MockMvcRequestBuilders
+                        .get("/faculty/" + id + "/students")
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$[*].id").value(id)); //выдает ошибку при сравнении 1 и 1
```

Comment:
потому что ты пытаешься взять id у всех элементов массива сразу, надо взять у какого-то одного, например, первого
```suggestion
                .andExpect(jsonPath("$[0].id").value(id)); //выдает ошибку при сравнении 1 и 1
```
а до этого лучше добавить проверку
`.andExpect(jsonPath("$.length()").value(1))`